### ISSUE TITLE
s-expandable-control enhancements

### ISSUE THREAD

**Issue by bnickel** [StackExchange/Stacks#269]
**Is your feature request related to a problem? Please describe.**
I have a button that shows and hides an editor panel.  I'm using `s-expandable-control` from the JavaScript package but have run into two issues.

1. There is no way to configure the button to indicate its state.
2. I want to be able to run a script every time it's about to show, in order to configure it.

**Describe the solution you'd like**
For 1, I'm planning on adding the `is-selected` to the source button.  It would like to toggle this class at the same time we're setting `aria-expanded`, along the lines of:

~~~javascript
if (this.element.classList.contains('s-btn')) {
    this.element.classList.toggle('is-selected', !newCollapsed);
}
~~~

For 2, I would like Stacks to emit events showing or hiding the element, something like `s-expandable-control:show` and `s-expandable-control:hide`.

**Describe alternatives you've considered**
For both 1 and 2, I could attach a click separate listener to the button to perform the changes, but I couldn't guarantee execution order so I'd have to use `setTimeout(_, 0)` or something. 

For 1, we could generalize the classes applied with `data-s-expandable-control-expanded-classes`, but that could be overkill.

**Comment by balpha** [comment]
#270 adds the events as you suggested, which should allow you to achieve 1) and 2) for the time being.

I don't think toggling `is-selected` purely because the element happens to be `s-btn` is a safe idea. For now I would suggest that you do this in your event handler.

Something like what you suggest in your last paragraph sounds better, but I would postpone this until [Stimulus gets the `classes` API](https://github.com/stimulusjs/stimulus/pull/202), because that's addressing exactly this use case.

**Comment by balpha** [comment]
We also added the `-toggle-class` parameter to #270, which has been merged now and released as part of [0.36.0](https://github.com/StackExchange/Stacks/releases/tag/v0.36.0).
